bonus_guides/G_static_assets.md

```diff
@@ -1 +1,289 @@
+
+
+#### Legacy Non-modularized Code
+
+If we have some legacy Javascript code which doesn't play well with module systems and/or we need global variables it defines, all we need to do is place our Javascript into directory `web/static/vendor`. Brunch will not wrap these files in modules.
+
+Let's test it. Create `web/static/vendor` if it does not exist yet and create file `web/static/vendor/meaning_of_life.js` with just one line in it:
+
+```
+meaning_of_life = 42;
+```
+
+Reload the page. Open the JS console and type `meaning_of_life`. This will return `42`. The variable is global.
+
+Important detail: according to the default configuration there is no ES6 support for files in `web/static/vendor`. Should we need to enable it, look for `plugins: { babel: { ignore:` in `brunch-config.js`.
+
+#### Brunch Plugin Pipeline
+
+All transformations Brunch performs are actually done by plugins. Brunch automatically uses installed plugins listed in  `package.json`. Here is what the pipeline looks like for a  newly generated Phoenix project:
+
+##### Javascript
+
+- [`babel-brunch`](https://github.com/babel/babel-brunch) transpiles ES6 code to vanilla ES5 Javascript using [Babel](https://github.com/babel/babel)
+- [`javascript-brunch`](https://github.com/brunch/javascript-brunch) is the main Javascript processing plugin. Without it Javascript files will be ignored.
+- [`uglify-js-brunch`](https://github.com/brunch/uglify-js-brunch) minifies Javascript code
+
+##### CSS
+
+- [`clean-css-brunch`](https://github.com/brunch/clean-css-brunch) is a minifier for CSS
+- [`css-brunch`](https://github.com/brunch/css-brunch) is the main CSS processing plugin. Without it  CSS files will be ignored.
+
+
+It is very easy to add more plugins. We can find a plethora of Brunch plugins [on the Brunch website](http://brunch.io/plugins.html) and [among NPM packages](https://www.npmjs.com/search?q=brunch).
+
+The order in which plugins run is defined by the order in which they appear in `package.json`.
+
+Let's add support for CoffeeScript. Edit `package.json` by adding the following line **before** `javascript-brunch`:
+
+```json
+  "coffee-script-brunch": ">= 1.8",
+```
+
+and run
+
+```
+npm install
+```
+
+Let's rename `greeter.js` into `greeter.coffee` and modify its contents to look like the following:
+
+```coffeescript
+Greet =
+  greet: -> console.log("Hello!")
+
+Bye =
+  greet: -> console.log("Bye!")
+
+module.exports =
+  Greet: Greet
+  Bye: Bye
+```
+
+Once Brunch has built our assets, reload our page, and we should see `Hello!` in the browser Javascript console, just like before.
+
+Adding support for other languages like SASS or React's JSX is as simple as this. Some plugins can be configured in `brunch-config.js`, but they will all work out of the box once installed.
```

Comment:
Since we include Babel out of the box, we get JSX support for free, out of the box.